src/Conway.hs

```diff
@@ -14,58 +13,53 @@ module Conway
   ) where
 
 import Data.Functor.Compose (Compose(..))
-import qualified Data.Vector as V
+import Data.Vector (Vector, (!), generate)
 import Data.Bool (bool)
 import Data.Distributive (Distributive(..))
 import Data.Functor.Rep (Representable(..), distributeRep)
 import Data.Functor.Identity (Identity(..))
-import Control.Arrow ((***))
 import Control.Comonad.Representable.Store (Store(..), StoreT(..), store, experiment)
 import Control.Comonad (Comonad(..))
 
 type Coord = (Int, Int)
-type Grid a = Store (Compose VBounded VBounded) a
-type Rule = Grid Bool -> Bool
+type Grid = Store (Compose Vector Vector) Bool
+type Rule = Grid -> Bool
 
-newtype VBounded a = VBounded (V.Vector a)
-  deriving (Eq, Show, Functor, Foldable)
-
-instance Distributive VBounded where
+instance Distributive Vector where
   distribute = distributeRep
 
-instance Representable VBounded where
-  type Rep VBounded = Int
-  index (VBounded v) i = v V.! (i `mod` gridSize)
-  tabulate desc = VBounded $ V.generate gridSize desc
+instance Representable Vector where
```

Comment:
I hadn't heard of those before, after some brief googling I can see why. I do think for a small project it's not such a big deal.